rpa-home/modules/ROOT/pages/invoke-rpa-from-flow-run.adoc

```diff
@@ -0,0 +1,106 @@
+
+# Use a published RPA Process in Salesforce Flow
+
+Publishing an invocable run configuration in RPA Manager, creates an External MuleSoft Service in Salesforce. The service's name consists of the name of the RPA Manager automation project without the blanks, followed by a '10'. 
+
+Follow these steps to use your process in a Salesforce Flow:
+
+. Create a new flow. In this example we use a screen flow for illustrational purposes.
+. Create a resource for the process execution input:
++ 
+image:invocableflow-resource-processExecution.png[Resource for the Process Execution]
++
+[cols="1,2"]
+|===
+|*Field*
+|*Value*
+
+|Resource Type
+|Select *Variable*.
+
+|API Name
+|Choose a name for this resource; for example `ProcessParameters`.
+
+|Data Type
+|Select *Apex-Defined*.
+
+|Apex Class
+|Select the Apex class *ExternalService__<RPA Project Name>10_ProcessExecution*, for example `ExternalService__ExampleProcess10_ProcessExecution``, 
+|===
+
+. Use a screen to set the input parameters for debugging purposes:
++
+image:invocableflow-screen-inputvariables.png[]
+
+. Assign all required values to the process parameters:
++ 
+image:invocableflow-assignment-processExecution.png[]
++
+The variable names are the same as the Activity Parameter names defined in RPA Builder:
++
+image:invocableflow-activityparameters.png[]
+
+. Choose the *Start Process* Action. You can filter for your process by name.
++
+[cols="1,2"]
+|===
+|*Process Start Type*
+|*Application*
+
+|*Start Process V2*
+|Ensures that only one instance of the process is running at the same time.
+
+|*Start Process Non Idempotent V2*
+|Enables parallel execution of the same process.
+
+|===
++
+image:invocableflow-action-choosestarttype.png[]
+
+. Configure the action filling the body with the resource you defined:
++
+image:invocableflow-action-startconfiguration.png[]
+
+. Get the execution Status for the first time:
++
+image:invocableflow-action-getfirstexecutionstatus.png[]
+
+. Create a resource for the status of the process execution:
++
+image:invocableflow-resource-processExecutionStatus.png[]
+
+. Create a resource for collecting all states the started process returns while running:
++
+image:invocableflow-resource-processExecutionStatusCollection.png[]
+
+. Add the first returned status to the status collection:
++
+image:invocableflow-assignment-processExecutionStatus1.png[]
+
+. Since the run time of a process is not known, add a loop over all returned status:
++
+image:invocableflow-loop.png[]
+
+. To check for the current status, define three text constants: `ìsStarted`, `running`, `error`.
+. Add a decision element:
++
+image:invocableflow-decision-checksucces.png[]
+
+. If the returned status is not success, add the current status to the loop collection:
++
+image:invocableflow-assignment-processExecutionStatusN.png[]
++
+This is repeated until the proces has finished and the status is success.
```

Comment:
```suggestion
This repeats until the process finishes 
with a status of `Success`.
```

Reply:
What happens if there is an error? Does the process keep looping or does it end?

Reply:
I don't know. I guess there has to be some kind of error handling. We have to check this with Salesforce developers.